### PR TITLE
Minor fixes to work on stable

### PR DESCRIPTION
**Pull request by Byron**
Tested on rustc 1.1.0 (35ceea399 2015-06-19)

### CODE REVIEW

**Comment by autohuonw** [comment]
Thanks for the pull request, and welcome! You should hear from @huonw (or someone else) soon.

File: src/main.rs [hunk 1]
```diff
@@ -35,7 +35,7 @@ struct DepInfo {
 
 // shallowly download the index, if necessary
 fn fetch_index() {
-    if std::path::PathBuf::new("crates.io-index").exists() {
+    if metadata(std::path::PathBuf::from("crates.io-index")).is_ok() {
```
Comment (huonw) [inline src/main.rs]: I think the old code would be better as `std::path::Path::new("crates.io-index")` instead of `PathBuf`.

**Comment by Byron** [comment]
Good point ! And it looks much better as well !

File: src/main.rs [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 extern crate glob;
-extern crate "rustc-serialize" as rustc_serialize;
+extern crate rustc_serialize;
 
 use std::collections::{HashMap, HashSet, hash_map};
 use std::io::prelude::*;
 use std::io::BufReader;
 use std::fs::File;
 use std::process::Command;
+use std::fs::metadata;
```
Comment (huonw) [inline src/main.rs]: Oh, could you instead import this via `use std::fs::{self, File};` two lines up?

File: src/main.rs [hunk 3]
```diff
@@ -35,7 +35,7 @@ struct DepInfo {
 
 // shallowly download the index, if necessary
 fn fetch_index() {
-    if std::path::PathBuf::new("crates.io-index").exists() {
+    if metadata("crates.io-index").is_ok() {
```
Comment (huonw) [inline src/main.rs]: and use this as `fs::metadata(...)`

**Comment by Byron** [comment]
And it's fixed ! Thanks for your patience.

**Comment by huonw** [comment]
Thanks for yours!

**Pull request merged**